File: Example/ObjC/Bulletin/CollectionViewWrapper.m
/**
 *  BulletinBoard
 *  Copyright (c) 2017 Alexis Aubry. Licensed under the MIT license.
 */

#import "CollectionViewWrapper.h"

@interface CollectionViewWrapper ()

@property (nonatomic, nonnull, strong, readwrite) UICollectionView *collectionView;

@end

@implementation CollectionViewWrapper

- (instancetype)initWithCollectionView:(UICollectionView *)collectionView
{
    self = [super initWithFrame:CGRectZero];
    if (self) {
        self.contentHeight = 256.0;
        self.collectionView = collectionView;
        [self addSubview:collectionView];
        collectionView.translatesAutoresizingMaskIntoConstraints = NO;

        [collectionView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor].active = YES;
        [collectionView.trailingAnchor constraintEqualToAnchor:self.trailingAnchor].active = YES;
        [collectionView.topAnchor constraintEqualToAnchor:self.topAnchor].active = YES;
        [collectionView.bottomAnchor constraintEqualToAnchor:self.bottomAnchor].active = YES;

        [self setContentHuggingPriority:UILayoutPriorityDefaultLow forAxis:UILayoutConstraintAxisHorizontal];
        [self setContentCompressionResistancePriority:UILayoutPriorityRequired forAxis:UILayoutConstraintAxisVertical];
    }
    return self;
}

- (void)setContentHeight:(CGFloat)contentHeight
{
    _contentHeight = contentHeight;
    [self invalidateIntrinsicContentSize];
}

- (CGSize)intrinsicContentSize
{
    // use 0 to have a flexible width, and use the fixed height
    return CGSizeMake(0, self.contentHeight);
}

@end

File: Example/Swift/Supporting Files/CollectionViewWrapper.swift
/**
 *  BulletinBoard
 *  Copyright (c) 2017 Alexis Aubry. Licensed under the MIT license.
 */

import UIKit

class CollectionViewWrapper: UIView {

    let collectionView: UICollectionView

    // change this value to the height you want for your collection view
    var pageHeight: CGFloat = 256

    init(collectionView: UICollectionView) {
        self.collectionView = collectionView
        super.init(frame: .zero)

        // pin the collection view to the edges of the wrapper
        addSubview(collectionView)
        collectionView.translatesAutoresizingMaskIntoConstraints = false
        collectionView.leadingAnchor.constraint(equalTo: leadingAnchor).isActive = true
        collectionView.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
        collectionView.topAnchor.constraint(equalTo: topAnchor).isActive = true
        collectionView.bottomAnchor.constraint(equalTo: bottomAnchor).isActive = true

        // allow the collection view to be resized to fit the width of the bulletin
        setContentHuggingPriority(.defaultLow, for: .horizontal)

        // do not allow the collection view to be resized vertically
        setContentCompressionResistancePriority(.required, for: .vertical)

    }

    @available(*, unavailable, message: "Use init(collectionView:)")
    required init?(coder aDecoder: NSCoder) {
        fatalError("Use init(collectionView:)")
    }

    override var intrinsicContentSize: CGSize {
        // use 0 to have a flexible width, and use the fixed height
        return CGSize(width: 0, height: pageHeight)
    }

}
